packages/main/src/plugin/index.ts

```diff
@@ -1131,6 +1144,13 @@ export class PluginSystem {
       return menuRegistry.getContributedMenus(context);
     });
 
+    this.ipcHandle(
+      'kube-generator-registry:getKubeGeneratorsIdsByType',
+      async (_, type: KubernetesGeneratorType): Promise<string[]> => {
+        return kubeGeneratorRegistry.getKubeGeneratorsIdsByType(type);
+      },
+    );
+
```

Comment:
it should return an array of KubeGeneratorsInfo objects

in the Info, it should contains the id/name but also the supported types.

Reply:
Why adding a name property ? For the UI ? Can't we just use the id for that ?

Reply:
so usually id is generated like a counter and name is provided by the provider

from the UI, you want to display `Do you want to generate using the Kompose Generator` ? and then you'll submit the id (like 0, 1 or uuid) to identify the resource

if we have only ids you'll display `Do you want to generate using the 12aa16 Generator ?`

Reply:
Why would we have the id generated here ? You want to remove the possibility for an extension to choose the id and provide only the name ?

Reply:
an example here
https://github.com/containers/podman-desktop/blob/main/packages/main/src/plugin/provider-registry.ts#L173-L182

(dumb approach)